--- a/android_webview/browser/aw_contents.cc
+++ b/android_webview/browser/aw_contents.cc
@@ -70,7 +70,6 @@
 #include "base/trace_event/trace_event.h"
 #include "base/trace_event/typed_macros.h"
 #include "components/adblock/content/browser/adblock_webcontents_observer.h"
-#include "components/adblock/content/browser/factories/adblock_telemetry_service_factory.h"
 #include "components/adblock/content/browser/factories/element_hider_factory.h"
 #include "components/adblock/content/browser/factories/resource_classification_runner_factory.h"
 #include "components/adblock/content/browser/factories/session_stats_factory.h"
@@ -279,8 +278,6 @@
   auto* browser_context = android_webview::AwBrowserContext::GetDefault();
   adblock::ResourceClassificationRunnerFactory::GetForBrowserContext(
       browser_context);
-  adblock::AdblockTelemetryServiceFactory::GetForBrowserContext(
-      browser_context);
   adblock::SessionStatsFactory::GetForBrowserContext(browser_context);
   adblock::SitekeyStorageFactory::GetForBrowserContext(browser_context);
   AdblockWebContentObserver::CreateForWebContents(
